### PR TITLE
Add ut for execution controller

### PR DESCRIPTION
**Pull request by jwcesign**
Signed-off-by: jwcesign <jiangwei115@huawei.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Add ut for execution controller

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

**Comment by jwcesign** [comment]
cc @Poor12 @RainbowMango

File: pkg/controllers/execution/execution_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+				mockInformerManager.EXPECT().GetSingleClusterManager(gomock.Any()).Return(mockSingleClusterInformerManager)
+				mockSingleClusterInformerManager.EXPECT().IsInformerSynced(gomock.Any()).Return(true)
+				mockSingleClusterInformerManager.EXPECT().Lister(gomock.Any()).Return(lister)
+				mockObjectWatcher.EXPECT().Delete(gomock.Any(), gomock.Any()).Return(nil)
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (jwcesign) [inline pkg/controllers/execution/execution_controller_test.go:103]: It's too much code, will add new case after this merged

File: pkg/util/objectwatcher/objectwatcher.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+//go:generate mockgen -source=objectwatcher.go  -destination=testing/mock_objectwatcher.go -package=testing ObjectWatcher
```
Comment (jwcesign) [inline pkg/util/objectwatcher/objectwatcher.go:1]: refering to k8s: https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/container/testing

**Comment by RainbowMango** [comment]
Please assign it to me once it's ready.

**Comment by jwcesign** [comment]
move to:https://github.com/karmada-io/karmada/pull/2640

**Pull request closed without merging**